okhttp/src/main/java/okhttp3/internal/cache/CacheStrategy.kt

```diff
@@ -13,314 +13,320 @@
-        this.receivedResponseMillis = cacheResponse.receivedResponseAtMillis();
-        Headers headers = cacheResponse.headers();
-        for (int i = 0, size = headers.size(); i < size; i++) {
-          String fieldName = headers.name(i);
-          String value = headers.value(i);
-          if ("Date".equalsIgnoreCase(fieldName)) {
-            servedDate = HttpDate.parse(value);
-            servedDateString = value;
-          } else if ("Expires".equalsIgnoreCase(fieldName)) {
-            expires = HttpDate.parse(value);
-          } else if ("Last-Modified".equalsIgnoreCase(fieldName)) {
-            lastModified = HttpDate.parse(value);
-            lastModifiedString = value;
-          } else if ("ETag".equalsIgnoreCase(fieldName)) {
-            etag = value;
-          } else if ("Age".equalsIgnoreCase(fieldName)) {
-            ageSeconds = UtilKt.toNonNegativeInt(value, -1);
+        this.sentRequestMillis = cacheResponse.sentRequestAtMillis()
+        this.receivedResponseMillis = cacheResponse.receivedResponseAtMillis()
+        val headers = cacheResponse.headers()
+        for (i in 0 until headers.size()) {
+          val fieldName = headers.name(i)
+          val value = headers.value(i)
+          when {
+            fieldName.equals("Date", ignoreCase = true) -> {
+              servedDate = HttpDate.parse(value)
+              servedDateString = value
+            }
+            fieldName.equals("Expires", ignoreCase = true) -> {
+              expires = HttpDate.parse(value)
+            }
+            fieldName.equals("Last-Modified", ignoreCase = true) -> {
+              lastModified = HttpDate.parse(value)
+              lastModifiedString = value
+            }
+            fieldName.equals("ETag", ignoreCase = true) -> {
+              etag = value
+            }
+            fieldName.equals("Age", ignoreCase = true) -> {
+              ageSeconds = value.toNonNegativeInt(-1)
+            }
           }
         }
       }
     }
 
-    /**
-     * Returns a strategy to satisfy {@code request} using the a cached response {@code response}.
-     */
-    public CacheStrategy get() {
-      CacheStrategy candidate = getCandidate();
+    /** Returns a strategy to satisfy [request] using [cacheResponse]. */
+    fun compute(): CacheStrategy {
+      val candidate = computeCandidate()
 
+      // We're forbidden from using the network and the cache is insufficient.
       if (candidate.networkRequest != null && request.cacheControl().onlyIfCached()) {
-        // We're forbidden from using the network and the cache is insufficient.
-        return new CacheStrategy(null, null);
+        return CacheStrategy(null, null)
       }
 
-      return candidate;
+      return candidate
     }
 
     /** Returns a strategy to use assuming the request can use the network. */
-    private CacheStrategy getCandidate() {
+    private fun computeCandidate(): CacheStrategy {
```

Comment:
renamed from getCandidate() to computeCandidate() ’cause Kotlin really likes get to be for accessors.